generated/niscope/niscope/session.py

```diff
@@ -2046,19 +2041,24 @@ def fetch_array_measurement(self, array_meas_function, timeout=hightime.timedelt
         '''
 
         meas_wfm, wfm_info = self._fetch_array_measurement(array_meas_function, timeout)
-        record_length = int(len(meas_wfm) / len(wfm_info))
 
-        for i in range(len(wfm_info)):
-            start = i * record_length
-            end = start + wfm_info[i]._actual_samples
-            wfm_info[i]._actual_samples = None
-            wfm_info[i].samples = meas_wfm[start:end]
+        record_length = int(len(meas_wfm) / len(wfm_info))
+        self._populate_samples_info(wfm_info, meas_wfm, record_length)
 
         num_records = int(len(wfm_info) / len(self._repeated_capability_list))
         self._populate_channel_and_record_info(wfm_info, self._repeated_capability_list, range(num_records))
 
         return wfm_info
 
+    def _populate_samples_info(self, waveform_info, sample_data, num_samples_per_item):
```

Comment:
`self` is never used.
Is that true for `_populate_channel_and_record_info` as well?

I wonder if they should both move to be free-standing functions in `waveform_info.py`.

Reply:
Good point. `self` isn't used there either. I like moving them to `waveform_info.py`. Only issue is that `_populate_channel_and_record_info` is needed for `measurement_stats` too. Do we create two copies of it, one for each type? Maybe that's ok.

Reply:
Have measurement_stats import waveform_info.py or viceversa? These are both fully handcoded. It also allows for adding unit tests to these functions.

Reply:
Yup, that works. Updated the PR.